internal/pkg/deploy/pipeline.go

```diff
@@ -82,78 +85,84 @@ type Source struct {
 // Secrets manager, which stores the GitHub Personal Access token if the
 // provider is "GitHub". Otherwise, it returns the detected provider.
 func (s *Source) GitHubPersonalAccessTokenSecretID() (string, error) {
-	// TODO type check if properties are GitHubProperties?
-	secretID, exists := s.Properties[manifest.GithubSecretIdKeyName]
-	if !exists {
-		return "", errors.New("the GitHub token secretID is not configured")
-	}
-
-	id, ok := secretID.(string)
-	if !ok {
-		return "", fmt.Errorf("unable to locate the GitHub token secretID from %v", secretID)
-	}
+	id := "N/A"
+	var ok bool
+	if s.ProviderName == manifest.GithubProviderName {
+		secretID, exists := s.Properties[manifest.GithubSecretIdKeyName]
+		if !exists {
+			return "", errors.New("the GitHub token secretID is not configured")
+		}
 
-	if s.ProviderName != manifest.GithubProviderName {
-		return fmt.Sprintf("Non-GitHub provider detected: %s", s.ProviderName), nil
+		id, ok = secretID.(string)
+		if !ok {
+			return "", fmt.Errorf("unable to locate the GitHub token secretID from %v", secretID)
+		}
 	}
 
 	return id, nil
 }
 
-type ownerAndRepo struct {
-	owner string
-	repo  string
-}
-
-func (s *Source) parseOwnerAndRepo() (*ownerAndRepo, error) {
-	if s.ProviderName != manifest.GithubProviderName {
-		return nil, fmt.Errorf("invalid provider: %s", s.ProviderName)
+// parseOwnerAndRepo parses the owner (if GitHub is the provider) and repo name from the repo URL.
+func (s *Source) parseOwnerAndRepo(provider string) (string, string, error) {
+	var repoExp *regexp.Regexp
+	if provider == manifest.GithubProviderName {
+		repoExp = ghRepoExp
+	}
+	if provider == manifest.CodeCommitProviderName {
+		repoExp = ccRepoExp
 	}
```

Comment:
Maybe define `repoExp` in the nearest place where it is firstly used.

Reply:
Done.